src/settings/TranslatorSettingsDialog.cpp

```diff
@@ -198,3 +215,46 @@ void TranslatorSettingsDialog::on_deleteRepo_clicked()
     repositoryModel_.removeRows(selection);
 }
 
+
+void TranslatorSettingsDialog::on_downloadButton_clicked()
+{
+    if (ui_->localModelTable->selectionModel()->selectedRows().size() != 1) {
+        // Can only download one model at a time for now
+        QMessageBox::warning(this, tr("Warning"), "Can only download one model at a time for now. Please select just one model.");
+    } else {
+        Model model = modelManager_->data(ui_->localModelTable->selectionModel()->selectedIndexes().first(), Qt::UserRole).value<Model>();
+        emit downloadModel(model);
+        this->setVisible(false); // Hide the settings window so we can see the download.
+    }
+}
+
+
+void TranslatorSettingsDialog::on_getMoreButton_clicked()
+{
+    modelManager_->fetchRemoteModels();
+    ui_->getMoreButton->setEnabled(false);
+}
+
+
+void TranslatorSettingsDialog::on_updateButton_clicked()
+{
+    if (ui_->localModelTable->selectionModel()->selectedRows().size() != 1) {
+        // Can only download one model at a time for now
+        QMessageBox::warning(this, tr("Warning"), "Can only download one model at a time for now. Please select just one model.");
+    } else {
+        Model oldmodel = modelManager_->data(ui_->localModelTable->selectionModel()->selectedIndexes().first(), Qt::UserRole).value<Model>();
+        // Find the model in the updated models list
+        auto todownload = modelManager_->findModelForUpdate(oldmodel);
+
+        if (todownload) {
+            emit downloadModel(todownload.value());
+            this->setVisible(false); // Hide the settings window so we can see the download.
```

Comment:
I suspect it makes more sense to show download progress either in a separate window (like Finder or any file manager when copying files) or in the settings window itself. But for now, while only allowing downloading *one* model, this is still acceptable behaviour? Especially because then it also selects that model and is ready to be used, which is nice.

Reply:
We can make batch downloading at a later point I think. My goal is to make this window a one-stop model management, so we can then refactor the main window without caring as much about how to display models for download (even if we eventually need to account for that)

Reply:
That sounds good to me.